### PR TITLE
New API endpoint: /registeredrelays

### PR DESCRIPTION
**Pull request by Marketen**
all relayers checked are predefined in the code
closes: https://github.com/dappnode/mev-sp-oracle/issues/28

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -98,6 +100,11 @@ type httpOkProofs struct {
 	RegisteredValidators   []uint64 `json:"registered_validators"`
 }
 
+type httpOkRelayersState struct {
+	RegisteredRelayers   string `json:"registered_relayers"`
```
Comment (alrevuelta) [inline api/api.go:104]: Add discussed, can we also have a `Registered` field that stores `true` or `false`?

File: api/api.go [hunk 2]
```diff
@@ -376,7 +384,61 @@ func (m *ApiService) handleValidatorOffchainStateByIndex(w http.ResponseWriter,
...
 }
 
+func (m *ApiService) handleValidatorRelayers(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	valPubKey := vars["valpubkey"]
+	if !IsValidPubkey(valPubKey) {
+		m.respondError(w, http.StatusInternalServerError, fmt.Sprintf("invalid validator pubkey format"))
+		return
+	}
+	var relays = []string{
```
Comment (alrevuelta) [inline api/api.go:394]: Can we move all this to `config.go`? And then use the `network` field to choose the relay set?

You would need to make a minor modification in `ApiService` struct, including the `Config` field. So that you can `m.Config.Network` or something like that.

And very important, we should differenciate between goerli and mainnet relayers. I have the feeling that here they are mixed (eg relay.ultrasound.money is for mainnet right?)

In other words, we need to know if the validator is registered in a goerli relayer OR mainnet relayer, depending on the network we are running. If mainnet, we only chain mainet relayer, if goerli we only check goerli relayers. We should not mix

File: api/api.go [hunk 3]
```diff
@@ -376,7 +384,61 @@ func (m *ApiService) handleValidatorOffchainStateByIndex(w http.ResponseWriter,
...
+		"mainnet-relay.securerpc.com",
+		"relayooor.wtf",
+		"relay.ultrasound.money",
+		"agnostic-relay.net",
+		"aestus.live",
+		"builder-relay-goerli.flashbots.net",
+	}
+	var registeredRelays []string
+	//var registeredRelaysString bytes.Buffer
+	var regRelayString strings.Builder
```
Comment (alrevuelta) [inline api/api.go:410]: lets use `var regRelayString []strings`. See my other comment in `httpOkRelayersState`

File: api/api.go [hunk 4]
```diff
@@ -98,6 +100,11 @@ type httpOkProofs struct {
 	RegisteredValidators   []uint64 `json:"registered_validators"`
 }
 
+type httpOkRelayersState struct {
+	RegisteredRelayers   string `json:"registered_relayers"`
+	UnRegisteredRelayers string `json:"unregistered_relayers"`
```
Comment (alrevuelta) [inline api/api.go:105]: both should be `[]string` as they are arrays.

**Review by alrevuelta — CHANGES_REQUESTED** [review]
looks good, some comments to address

File: api/api.go (hunk 2 above)
Comment (Marketen) [reply api/api.go]: relays changed to config.go. 2 sets, one for mainnet and one for goerli. 

ApiService now has a new field "Network", initialized at "NewApiService" method. The endpoint handler of /registeredrelays/{pubkey} now checks the Network field to choose between the two relay sets.

Related to this, I did a minor extra change at config.go, setting "mainnet" as a default value when -network flag is not provided by user.

File: config/config.go [hunk 5]
```diff
@@ -15,88 +18,154 @@ type Config struct {
...
-
-	// Debug flags, never use in production
-	DebugHardcodedSubscriptions []uint64
 }
 
 // By default the release is a custom build. CI takes care of upgrading it with
 // go build -v -ldflags="-X 'github.com/xxx/yyy/config.ReleaseVersion=x.y.z'"
 var ReleaseVersion = "custom-build"
 
+var MainRelays = []string{
```
Comment (alrevuelta) [inline config/config.go:35]: minor thing, can we use `const`. would be safer :)

**Review by alrevuelta — APPROVED** [review]
